scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/DatasetController.java

```diff
@@ -0,0 +1,191 @@
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+import java.util.List;
+
+
+@Controller
+public class DatasetController {
+
+    private final DatasetService datasetService;
+    private final DataFileService dataFileService;
+    private final IDatasetService datasetManager;
+    private final IUserManager userManager;
+
+    @Autowired
+    public DatasetController(DatasetService datasetService,
+                             DataFileService dataFileService,
+                             IDatasetService datasetManager,
+                             IUserManager userManager) {
+        this.datasetService = datasetService;
+        this.dataFileService = dataFileService;
+        this.datasetManager = datasetManager;
+        this.userManager = userManager;
+    }
+
+    @RequestMapping(value = "datasets/{username}", method = RequestMethod.GET)
+    public String get(Model model, @PathVariable String username) {
+        model.addAttribute("dataset", new DatasetForm());
+        List<IDataset> dbDatasets = datasetService.findAllByUsername(username);
+        if (dbDatasets.size() > 0) {
+            model.addAttribute("dbDatasets", dbDatasets);
+        } else {
+            model.addAttribute("noDatasets", "You Have No Datasets Yet");
+        }
+        return "datasets/index";
+    }
+
+    @RequestMapping(value = "datasets/add", method = RequestMethod.POST)
+    public String post(@Valid @ModelAttribute("dataset") DatasetForm datasetForm,
+                       BindingResult result,
+                       Model model,
+                       RedirectAttributes redirectAttributes) {
+        boolean success;
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        String username = authentication.getName();
+        IUser user = userManager.findByUsername(username);
+        datasetForm.setUsername(username);
+
+        if (result.hasErrors()) {
+            // Handle form error - They don't show up automatically with redirects
+            if (datasetForm.getName().isEmpty()) {
+                redirectAttributes.addFlashAttribute("noDatasetName", "Dataset Name Cannot Be Empty");
+            }
+            redirectAttributes.addFlashAttribute("dataset", datasetForm);
+            redirectAttributes.addAttribute("username", username);
+            return "redirect:/datasets/{username}";
+        }
+
+        try {
+            success = datasetService.createDataset(datasetForm.getName(), user);
+        } catch (DatasetException e) {
+            model.addAttribute("dataset", new DatasetForm());
+            redirectAttributes.addFlashAttribute("message", e.getMessage());
```

Comment:
Usually, you shouldn't show the exception to the user but show a proper error message based on the type of exception.